src/frame.js

```diff
@@ -0,0 +1,14 @@
+//each frame initializes with empty array
+
+function Frame(){
+  this.outcome = []
```

Comment:
The Bowl abstraction feels really useful.  However, it feels like Frame objects break the encapsulation of Bowl objects by copying each roll from bowl into outcome.